synthesizer/src/vm/mod.rs

```diff
@@ -69,6 +71,42 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
             }
         }
 
+        // Load all the deployments from the store.
+        Self::load_deployments(&mut process, &store)?;
+
+        // Return the new VM.
+        Ok(Self { process: Arc::new(RwLock::new(process)), store })
+    }
+
+    #[cfg(not(feature = "serial"))]
+    #[inline]
+    fn load_deployments(process: &mut Process<N>, store: &ConsensusStore<N, C>) -> Result<()> {
+        // Retrieve the transaction store.
+        let transaction_store = store.transaction_store();
+        // Retrieve the stacks of the deployments from the store.
+        let stacks = transaction_store
```

Comment:
Why not use the same code structure in both cases, by using `map` in both places? That way we can reuse the `cfg_iter` code everywhere.

Reply:
The issue is that we don't have direct parallel iteration for database records - there's an intermediate collection into a vector. I don't how this could be achieved without introducing clunky APIs.